spec/services/adviser/application_form_validations_spec.rb

```diff
@@ -0,0 +1,211 @@
+    end
+
+    it 'needs a first_name' do
+      application_form.first_name = nil
+      expect(validations).to have_error_on(:first_name)
+    end
+
+    it 'needs a last_name' do
+      application_form.last_name = nil
+      expect(validations).to have_error_on(:last_name)
+    end
+
+    it 'needs a date_of_birth' do
+      application_form.date_of_birth = nil
+      expect(validations).to have_error_on(:date_of_birth)
+    end
+
+    it 'needs a phone_number' do
+      application_form.phone_number = nil
+      expect(validations).to have_error_on(:phone_number)
+    end
+
+    it 'needs a country' do
+      application_form.country = nil
+      expect(validations).to have_error_on(:country)
+    end
+
+    it 'needs an applicable_degree' do
+      expect(validations).to have_error_on(:applicable_degree)
+    end
+
+    context 'when the candidate has a domestic degree' do
+      before do
+        create(:degree_qualification,
+               :adviser_sign_up_applicable,
+               application_form: application_form)
+      end
+
+      it 'needs a postcode' do
+        application_form.postcode = nil
+        expect(validations).to have_error_on(:postcode)
+      end
+
+      context 'when the candidate does not have Maths and English GCSEs' do
+        it 'has errors on the GCSE fields' do
+          expect(validations).to have_error_on(:maths_gcse)
+          expect(validations).to have_error_on(:english_gcse)
+        end
+      end
+
+      context 'when the candidate is missing and is not retaking their Maths and English GCSEs' do
+        before do
+          create(:gcse_qualification, :missing_and_not_currently_completing, subject: 'maths', application_form: application_form)
+          create(:gcse_qualification, :missing_and_not_currently_completing, subject: 'english', application_form: application_form)
+        end
+
+        it 'has errors on the GCSE fields' do
+          expect(validations).to have_error_on(:maths_gcse)
+          expect(validations).to have_error_on(:english_gcse)
+        end
+      end
+
+      context 'when the candidate is missing, but currently retaking their Maths and English GCSEs' do
+        before do
+          create(:gcse_qualification, :missing_and_currently_completing, subject: 'maths', application_form: application_form)
+          create(:gcse_qualification, :missing_and_currently_completing, subject: 'english', application_form: application_form)
+        end
+
+        it 'does not have errors on the GCSE fields' do
+          expect(validations).not_to have_error_on(:maths_gcse)
+          expect(validations).not_to have_error_on(:english_gcse)
+        end
+      end
+
+      context 'when the application form is valid' do
+        let(:application_form) { create(:completed_application_form, :with_degree_and_gcses) }
+
+        it { is_expected.to be_valid }
+        it { expect(validations.applicable_degree).not_to be_international }
```

Comment:
Should this be in a different context?

Reply:
Yeah that should be in the parent context as its regardless of the valid state 👍